cpp/src/io/parquet/reader_impl_preprocess.cu

```diff
@@ -967,13 +972,31 @@ void reader::impl::load_and_decompress_data()
     task.wait();
   }
 
+  rmm::device_scalar<int32_t> error_code(0, _stream);
+
   // Process dataset chunk pages into output columns
-  auto const total_pages = count_page_headers(chunks, _stream);
+  auto const total_pages = count_page_headers(chunks, error_code.data(), _stream);
   if (total_pages <= 0) { return; }
+
+  auto decode_error = error_code.value(_stream);
```

Comment:
do we need to check the error twice?

Reply:
Well, the same kernel is called twice. I suppose we could initialize the error code to 0 outside the kernel, and then since errors are OR'd in we could check once. After both are called. Or just exit early from the second call if error is non zero.

Reply:
the error is initialized outside of the kernel. I think having a single check should work, as long as we're not worried about a segfault in decode_page_headers if there's an error in count_page_headers.

Reply:
hmm, so the code is initialized, so no worries there.  but if something goes wrong in count_page_headers and it returns 0, then we'll just return without the error. So we kind of have to check after count_page_headers.  Right now I don't think there's an error that won't happen in the first call, that can happen in the second, but down the road...

I think it's safest to just leave both in (unless it turns out that pulling back the error code is expensive).

Reply:
Let's assume error checking is cheap and keep both checkpoints:
1) Please move the common code to a util function (we got three copies now)?
2) Please run benchmarks with the new checks.